shortfin/tests/apps/llm/cpu_llm_server_test.py

```diff
@@ -0,0 +1,156 @@
+import pytest
+import subprocess
+import time
+import requests
+import os
+import json
+import uuid
+import tempfile
+import shutil
+
+from transformers import AutoTokenizer
+
+BATCH_SIZES = [1, 4]
+cpu_settings = {
+    "device_flags": ["-iree-hal-target-backends=llvm-cpu"],
+    "device": "local-task",
+}
+gpu_settings = {
+    "device_flags": ["-iree-hal-target-backends=rocm", "--iree-hip-target=gfx1100"],
+    "device": "hip",
+}
+settings = cpu_settings
+
+
+@pytest.fixture(scope="module")
+def model_test_dir(tmp_path_factory):
+    tmp_dir = tmp_path_factory.mktemp("cpu_llm_server_test")
+    try:
+        # Download model if it doesn't exist
+        model_path = os.path.join(tmp_dir, "open-llama-3b-v2-f16.gguf")
+        if not os.path.exists(model_path):
+            subprocess.run(
+                f"huggingface-cli download --local-dir {tmp_dir} SlyEcho/open_llama_3b_v2_gguf open-llama-3b-v2-f16.gguf",
+                shell=True,
+                check=True,
+            )
+
+        # Set up tokenizer if it doesn't exist
+        tokenizer_path = os.path.join(tmp_dir, "tokenizer.json")
+        if not os.path.exists(tokenizer_path):
+            tokenizer = AutoTokenizer.from_pretrained(
+                "openlm-research/open_llama_3b_v2"
+            )
+            tokenizer.save_pretrained(tmp_dir)
+
+        # Export model if it doesn't exist
+        mlir_path = os.path.join(tmp_dir, "model.mlir")
+        config_path = os.path.join(tmp_dir, "config.json")
+        if not os.path.exists(mlir_path) or not os.path.exists(config_path):
+            bs_string = ",".join(map(str, BATCH_SIZES))
+            subprocess.run(
+                [
+                    "python",
+                    "-m",
+                    "sharktank.examples.export_paged_llm_v1",
```

Comment:
At the moment, sharktank and shortfin are self-contained subprojects. The CI workflows build one _or_ the other, not both.

This is an integration test that uses both, so it should live outside of `shortfin/`. Two ideas for that:
* A new top level `tests/` directory for integration tests
* A new top level `shark_platform/` directory for the combination of all packages, like what I started on in https://github.com/nod-ai/SHARK-Platform/pull/244

Either way, we'll need a new workflow file instead of the existing `ci.*shortfin.*yml` or `ci.*sharktank.*yml` files.

See also https://github.com/nod-ai/SHARK-Platform/pull/284#discussion_r1803610173

> This can be separate from this PR, but we should use an in-memory API for all model exporting/importing workflows, rather than fork into a subprocess.

Reply:
Moved test to toplevel `tests` directory. Will fix existing CIs and implement new CI after responding to code feedback

Reply:
@stellaraccident do you have an opinion on where integration tests using both shortfin and sharktank should live?

* `tests/`
* `shark_platform/tests` (meta package _and_ integration tests?)

Reply:
Oooh wait. I already wrote a test and workflow for this, but that was using the old version of shortfin: https://github.com/nod-ai/SHARK-Platform/pull/70

We should replace https://github.com/nod-ai/SHARK-Platform/blob/main/build_tools/integration_tests/llama_export_compile_serve.sh and https://github.com/nod-ai/SHARK-Platform/blob/main/.github/workflows/test.yaml with this new code using the latest APIs :P

That workflow has been failing (since it wasn't updated to use the new APIs): https://github.com/nod-ai/SHARK-Platform/actions/workflows/test.yaml

Reply:
Prior discussion about paths was https://github.com/nod-ai/SHARK-Platform/pull/70#discussion_r1655281497

> Let's put these scripts in build_tools/integration_tests. Python projects end up using the tests/ folder for more unit-testy things.

Reply:
Moved test to `./build_tools/integration_tests/cpu_llm_server_test.py`, removed `test.yaml` and `llama_export_and_compile_serve.sh`

Reply:
@ScottTodd Workflow file added and passing. 

In regard to: 
```See also https://github.com/nod-ai/SHARK-Platform/pull/284#discussion_r1803610173```

This integration test was in part spurred by a change being implemented in the sharktank export script, which caused a failure when using exported models to launch the shortfin llm server. So the export script, in this case, would be considered part of the "e2e" flow for testing the llm server.

With that in mind does it still make sense to switch to using `vllm.LLM` or is this a broader suggestion that maybe we shouldn't be using the export script as part of the "golden" flow?

Reply:
> With that in mind does it still make sense to switch to using `vllm.LLM` or is this a broader suggestion that maybe we shouldn't be using the export script as part of the "golden" flow?

The current export script is a developer tool built for a single model that has rough edges. We can improve or create something new. Users need something better either way.

Reply:
Gotcha, does this make sense for a subsequent PR? It sounds like a separate task from this one.

I know that raises some caution of the "can getting kicked down the road", so maybe we could create an issue for this right now and I can discuss in sync tomorrow to get it assigned?

Reply:
Yep, I just want to make sure we keep focused on providing a good user experience. We should have easy to use scripts and test _those_.

Reply:
Issue created here: https://github.com/nod-ai/SHARK-Platform/issues/402